src/field/field_data.cxx

```diff
@@ -6,8 +6,11 @@
 #include <field_factory.hxx>
 #include "unused.hxx"
 
-FieldData::FieldData() : boundaryIsCopy(false), boundaryIsSet(true) {
-  
+FieldData::FieldData(Mesh* m) :
+    fielddatamesh(m), boundaryIsCopy(false), boundaryIsSet(true) {
```

Comment:
I think this could be `fielddatamesh(m? m : mesh)` to avoid the need for the logic inside the function.

Reply:
I'd keep the explicit `nullptr` check though:

`fielddatamesh(m == nullptr ? m : mesh)`

Reply:
I've fixed this, and also moved it into the header -- the ctor body is now empty